src/components/SlateEditor/plugins/uuDisclaimer/SlateDisclaimer.tsx

```diff
@@ -0,0 +1,138 @@
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  position: absolute;
+  right: -${spacing.large};
+`;
+
+const StyledModalHeader = styled(ModalHeader)`
+  padding-bottom: 0px;
+`;
+
+const SlateDisclaimer = ({ attributes, children, element, editor }: Props) => {
+  const { t } = useTranslation();
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+
+  const embed: UuDisclaimerMetaData = useMemo(
+    () => ({
+      status: "success",
+      data: {},
+      embedData: element.data,
+      resource: element.data?.resource,
+    }),
+    [element.data],
+  );
+
+  const handleDelete = () => {
+    const path = ReactEditor.findPath(editor, element);
+    Transforms.removeNodes(editor, {
+      at: path,
+      match: (node) => Element.isElement(node) && node.type === TYPE_DISCLAIMER,
+    });
+    setTimeout(() => {
+      ReactEditor.focus(editor);
+      Transforms.select(editor, path);
+      Transforms.collapse(editor);
+    }, 0);
+  };
+
+  const handleRemoveDisclaimer = () => {
+    const path = ReactEditor.findPath(editor, element);
+    Transforms.unwrapNodes(editor, {
+      at: path,
+      match: (node) => Element.isElement(node) && node.type === TYPE_DISCLAIMER,
+    });
+    setTimeout(() => {
+      ReactEditor.focus(editor);
+      Transforms.select(editor, path);
+      Transforms.collapse(editor);
+    }, 0);
+  };
+
+  const onSaveDisclaimerText = useCallback(
+    (values: UuDisclaimerEmbedData) => {
+      setModalOpen(false);
+      ReactEditor.focus(editor);
+      const path = ReactEditor.findPath(editor, element);
+      Transforms.setNodes(
+        editor,
+        {
+          data: values,
+          isFirstEdit: false,
+        },
+        { at: path },
+      );
+      if (Editor.hasPath(editor, Path.next(path))) {
+        setTimeout(() => {
+          Transforms.select(editor, Path.next(path));
+        }, 0);
+      }
+    },
+    [setModalOpen, editor, element],
+  );
+
+  return (
+    <div data-testid="slate-disclaimer-block" {...attributes}>
+      <ButtonContainer>
+        <DeleteButton aria-label={t("delete")} data-testid="delete-disclaimer" onClick={handleDelete} />
+        <Modal open={modalOpen} onOpenChange={setModalOpen}>
+          <ModalTrigger>
+            <IconButtonV2 variant="ghost" aria-label={t("form.disclaimer.edit")} data-testid="edit-disclaimer">
+              <Pencil />
+            </IconButtonV2>
+          </ModalTrigger>
+          <ModalContent size="normal">
+            <StyledModalHeader>
+              <ModalTitle>{t("form.disclaimer.title")}</ModalTitle>
+              <ModalCloseButton />
+            </StyledModalHeader>
+            <DisclaimerForm initialData={embed?.embedData} onOpenChange={setModalOpen} onSave={onSaveDisclaimerText} />
+          </ModalContent>
+        </Modal>
+        <MoveContentButton aria-label={t("move")} data-testid="move-disclaimer" onMouseDown={handleRemoveDisclaimer} />
```

Comment:
Denne har mista teksten sin. 
```suggestion
        <MoveContentButton aria-label={t(""learningResourceForm.fields.rightAside.moveContent"")} data-testid="move-disclaimer" onMouseDown={handleRemoveDisclaimer} />
```

Reply:
Arrgh, lurt av Copilot 😠